frontend/src/store/modules/auth/actions.js

```diff
@@ -25,6 +27,19 @@ export default {
         });
     },
 
+    [UPDATE]: (context, data) => {
+        return new Promise((resolve, reject) => {
```

Comment:
not quite sure why you need this promise, `updateUser` already returns Promise, that you can use for chaining

or you can use async/await

in any case - user should be undefined here